docs/Minimal_Build__Advanced_Usage_with_NNAPI.md

```diff
@@ -1,77 +1,121 @@
-# ONNX Runtime Minimal Build: Advanced Usage with NNAPI
+# ONNX Runtime Mobile Build: Advanced Usage with NNAPI
+
+The ONNX Runtime Mobile with the NNAPI execution provider can be used to execute ORT format models on Android platforms. This document explains the different options available to optimize the ORT model for exection with NNAPI. 
+
+Review the details for using NNAPI with ONNX Runtime Mobile [here](docs/ONNX_Runtime_for_Mobile_Platforms.md#using-nnapi-with-onnx-runtime-mobile).
 
 ## Effect of Optimization Levels on NNAPI Execution Provider
 
-### Background
+The performance of the ORT model execution through NNAPI is influenced by the different optimization applicable when the ORT formatted model is created.
+
+### Background about optimization for ORT Models
 
-There are three main steps that affect the operators used in an optimized model, and which execution provider (EP) runs each node. 
+ONNX Runtime applies optimizations on the ONNX model to improve inference performance. There are three optimization steps:
 
-*Step 1:* 
+*Step 1:*
 
-The 'basic' optimizations are run. See the [graph optimization](ONNX_Runtime_Graph_Optimizations.md) documentation for further details. Only ONNX operators are used when modifying the model in this step. 
+The _basic_ optimizations (removing redundant nodes and computations) are applied on the ONNX model. See the [graph optimization](ONNX_Runtime_Graph_Optimizations.md) documentation for further details. Only ONNX operators are used when modifying the model in this step.
 
-*Step 2:* 
+*Step 2:*
 
-The enabled [execution providers](execution_providers/README.md) are asked which nodes they can execute. Each node is assigned to an EP, with the user defined priority determining which EP is selected. 
+The [execution providers](execution_providers/README.md) (EP) specific optimizations assign nodes to EPs based on the reponse from the EP. The user defined priority is used to allocate nodes among different hardware specific execution providers.
 
-When the NNAPI EP looks at the model, it identifies nodes that are using ONNX operators that can be executing using NNAPI. It will group together as many nodes as possible that can be executed by NNAPI at once, as there is a cost to copy data between the CPU and NNAPI. The more nodes in each group, and the fewer groups, the better the performance will be.
+Nodes in the ORT model are assigned to the NNAPI EP using this process.
+
+ONNX Runtime may group nodes into a sub-graph for efficient execution via NNAPI. This reduces the overhead to copy data between the CPU and NNAPI when executing the nodes. The inference performace is better with more nodes in each sub-graph, and fewer sub-graphs.
```

Comment:
nit: It's the EP implementation that does any grouping and not ONNX Runtime code.

Reply:
is the EP part of the ONNX Runtime lib? Code sits in the EP vs. Core but its not a separate library.

Reply:
my point is more that the EP implementation makes the decision (so if there's an issue with grouping it's the EPs fault) not ORT, so it's the NNAPI EP that decides which nodes form a group. the EP is not a separate library though. 

I mainly want the correct thing to be blamed if someone is filing an issue.